src/ThemeProvider.tsx

```diff
@@ -0,0 +1,33 @@
+import { Resolver } from '@stoplight/json-ref-resolver';
+import * as React from 'react';
+
+import { baseTheme } from './theme';
+import { IThemeInterface } from './types';
+import { ThemeProvider as StyledProvider } from './utils';
+
+const resolver = new Resolver({
+  getRef(_key, val) {
+    if (typeof val === 'string' && val.startsWith('#/')) return val;
+    return;
+  },
+});
+
+export interface IThemeProviderProps {
+  theme?: IThemeInterface;
+  children?: any;
+}
+export const ThemeProvider = (props: IThemeProviderProps) => {
```

Comment:
Suggest using state + only run resolver on mount/dismount for now. COULD pass in theme object to run resolver when it changes, not sure if needed though. 

Note, useEffect cannot be an async function, so I define the function outside of the effect (which is harmless).

Example:

```tsx
export const ThemeProvider = (props: IThemeProviderProps) => {
  const [theme, setTheme] = useState(Object.assign({}, baseTheme, props.theme));

  const resolveTheme = async () => {
    const resolved = await resolver.resolve(theme);
    
    if (resolved.errors && resolved.errors.length > 0) {
      console.error('Error resolving theme object.', resolved.errors)
      return;
    }

    setTheme(resolved.result);
  }

  React.useEffect(() => {
    resolveTheme();
  }, []);

  return <StyledProvider theme={theme}>{props.children}</StyledProvider>;
};
```

Reply:
If we only run on mount/unmount the ui won't get appropriately updated with the theme object changes

Reply:
A bit out of context, but that callback could be omitted I suppose, so ultimately the effect hook would be invoked in the following way:
```js
React.useEffect(resolveTheme, [props.theme]);
```

Reply:
I think that `useEffect` only supports a function that returns `void | () => void` (disposer). I don't think it supports returning a promise (which is what resolveTheme does).